lib/src/proto-to-firestore.ts

```diff
@@ -14,33 +14,35 @@
  * limitations under the License.
  */
 
-import { FieldDescriptor, MessageDescriptor, registry } from "./message-registry";
-import { DocumentData, DocumentFieldValue } from "@google-cloud/firestore";
+import {FieldDescriptor, MessageDescriptor, registry} from './message-registry';
+import {DocumentData, DocumentFieldValue} from '@google-cloud/firestore';
 
 /**
  * Returns the map representation of the provided message ready for serialization in Firestore.
- * The map is keyed by message field numbers represented as strings, while field values are 
- * converted to corresponding Firestore data types. 
+ * The map is keyed by message field numbers represented as strings, while field values are
+ * converted to corresponding Firestore data types.
  */
 export function toDocumentData(message: any): DocumentData | Error {
   const type = message.constructor.name;
-  const descriptor =  registry.getDescriptor(message);
+  const descriptor = registry.getMessageDescriptor(message.constructor);
+  // Fail if message not defined in registry.
   if (!descriptor) return Error(`Unknown message type ${type}`);
-
-  if (!descriptor.fields) {
-    // `values`, `nested`, and `oneofs` fields are handled implicitly in `messageToData()`.
-    return Error(`Unexpected field type ${descriptor}`);
-  }
+  // Messages must also have at least one field.
+  if (!descriptor.fields)
+    return Error(`Invalid message definition: ${descriptor}`);
   return messageToData(message, descriptor);
 }
 
-function messageToData(message: any, descriptor: MessageDescriptor): DocumentData {
+function messageToData(
+  message: any,
+  descriptor: MessageDescriptor
+): DocumentData {
   const firestoreMap: DocumentData = {};
   for (const name in descriptor.fields) {
     const fieldDescriptor = descriptor.fields[name];
     const fieldNumber = fieldDescriptor?.id;
     if (!fieldNumber) {
-      console.debug(`Skipping unknown field $name in $type`);
+      console.debug('Skipping unknown field $name in $type');
```

Comment:
is this what you want to print? 'Skipping unknown field $name in $type'

Reply:
Sorry, I got a case of Kotlin-brain. Fixed.